src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -768,6 +769,25 @@ object Parser {
         ghostPrimaryExp |
         operand
 
+    // TODO: change delimiter to { and implement required ambiguity resolution, current: !< X, ..., Z !>
+    // declaredPred!<d1, ..., dn!>
+    lazy val fpredConstruct: Parser[PPredConstructor] =
+      (idnUse ~ predConstructArgs) ^^ { p => PPredConstructor(PFPredBase(p._1), p._2) }
+
+    lazy val mpredConstruct: Parser[PPredConstructor] =
+      (expression <~ ".") ~ idnUse ~ predConstructArgs ^^ {
```

Comment:
Does `(expression <~ ".") ~ idnUse` work? I have the impression that `expression` might already parse the ".". You could try to replace this first part by `selection` which parses a `PDot`. Note that PDot allows a type in front of "." but it would be fine for me that the type checker currently does not support mpredicate constructors that take the receiver expression as an additional argument

Reply:
This was a local change that I did not intend to commit. Anyhow, thanks for the `selection` suggestion, that was helpful!